src/main/java/in/projecteka/gateway/registry/RegistryService.java

```diff
@@ -165,6 +167,11 @@
                             Endpoints endpoints = new Endpoints();
                             return Flux.fromIterable(services)
                                     .flatMap(request -> {
+                                        if(request.getEndpoints() != null && request.getEndpoints().size() > 0) {
+                                        if(request.getEndpoints().stream().anyMatch(endpoint ->
+                                                endpoint.getUse() == null || endpoint.getConnectionType() == null || !StringUtils.hasText(endpoint.getAddress()))) {
```

Comment:
```suggestion
                                                endpoint.getUse() == null || endpoint.getConnectionType() == null || !StringUtils.hasText(endpoint.getAddress()))) {
```

extract this as a method please. `isInvalidEndpoint` or something.